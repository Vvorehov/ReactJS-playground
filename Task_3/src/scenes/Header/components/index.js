export { FilmInfo } from './FilmInfo';
export { Search } from './Search';
export { HeaderTop } from './HeaderTop';