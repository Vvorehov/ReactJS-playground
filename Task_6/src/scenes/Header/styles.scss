#header {
  display: flex;
  align-items: center;
  justify-content: space-between;
  color: #ED5664;
  padding: 0 6rem 5rem;
}

.backToHome {
  color: #ED5664;
  padding: 0.5rem 1em;
  background: #fff;
  display: inline-block;
  border-radius: 2px;
}