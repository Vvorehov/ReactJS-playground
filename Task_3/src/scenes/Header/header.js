import React from "react";
import { render } from "react-dom";
import { Search } from "./components/Search"

const Header = () => {
    return <header>
        <h1>Find your movie</h1>
        <Search />
    </header>;
};

export default Header;