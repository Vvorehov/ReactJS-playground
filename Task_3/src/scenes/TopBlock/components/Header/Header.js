import React from "react";
import { Logo } from "../../../Logo";
import "./styles.scss";

const Header = () => {
  return <header id="header" className="header-top">
    <Logo siteName='netflixroulette'/>
    <a className="backToHome" href="/">Search</a>
  </header>;
};

export default Header;