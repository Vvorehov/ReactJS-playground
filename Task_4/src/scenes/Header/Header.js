import React from "react";
import { Logo } from "../Logo";
import "./styles.scss";

const Header = () => {
  return <header id="header">
    <Logo siteName="netflixroulette"/>
    <a className="backToHome" href="/">Search</a>
  </header>;
};

export default Header;