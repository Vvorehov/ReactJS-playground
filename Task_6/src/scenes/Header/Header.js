import React from "react";
import { Logo } from "../Logo";
import "./styles.scss";

const Header = (props) => {
  return <header id="header">
    <Logo siteName="netflixroulette"/>
    <a className={props.showSearchLink ? 'backToHome':'isHidden'} href="/">Search</a>
  </header>;
};

export default Header;