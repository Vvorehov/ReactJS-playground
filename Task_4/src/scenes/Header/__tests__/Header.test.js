import React from "react";
import Header from "../Header";
import TestRenderer from 'react-test-renderer';

describe('Header layout complete', () => {
    it('Render correct', () => {

        const component = TestRenderer.create(
            <Header />
        );

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

