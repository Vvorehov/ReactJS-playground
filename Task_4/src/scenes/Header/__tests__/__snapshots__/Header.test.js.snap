// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header layout complete Render correct 1`] = `
<header
  id="header"
>
  <div
    className="siteLogo"
  >
    netflixroulette
  </div>
  <a
    className="backToHome"
    href="/"
  >
    Search
  </a>
</header>
`;
